ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -453,10 +430,51 @@ def test_ipa_healthcheck_log_rotate_file_exist_issue35(self):
         tasks.install_packages(self.master, HEALTHCHECK_PKG)
         assert self.master.transport.file_exists(HEALTHCHECK_LOG_ROTATE_CONF)
         cmd = self.master.run_command(
-            ['logrotate', '--debug', HEALTHCHECK_LOG_ROTATE_CONF]
+            ["logrotate", "--debug", HEALTHCHECK_LOG_ROTATE_CONF]
         )
         assert msg not in cmd.stdout_text
 
+    def test_ipa_dns_systemrecords_check(self):
+        """
+        This test ensures that ipahealthcheck.ipa.idns check
+        displays the correct result when master and replica is setup along
+        with integrated DNS.
+        """
+        SRV_MASTER_RECORDS = [
+            "_ldap._tcp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kerberos._tcp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kerberos._udp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kerberos-master._tcp." + self.domain.name + ".:" + api.env.host +
+            ".",
+            "_kerberos-master._udp." + self.domain.name + ".:" + api.env.host +
+            ".",
+            "_kpasswd._tcp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kpasswd._udp." + self.domain.name + ".:" + api.env.host + ".",
+            '"' + self.domain.name.upper() + '"',
+            self.master.ip,
+        ]
+        SRV_REPLICA_RECORDS = [
+            "_ldap._tcp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kerberos._tcp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kerberos._udp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kerberos-replica._tcp." + self.domain.name +
+            ".:" + api.env.host + ".",
+            "_kerberos-replica._udp." + self.domain.name +
+            ".:" + api.env.host + ".",
+            "_kpasswd._tcp." + self.domain.name + ".:" + api.env.host + ".",
+            "_kpasswd._udp." + self.domain.name + ".:" + api.env.host + ".",
+            '"' + self.domain.name.upper() + '"',
+            self.replicas[0].ip,
+        ]
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.idns",
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] in SRV_MASTER_RECORDS
+            assert check["kw"]["key"] in SRV_REPLICA_RECORDS
```

Comment:
It may work if you change to an or instead. Any given key will only be for one of the records.

Reply:
@rcritten have made the change